framework/src/meshgenerators/PatternedMeshGenerator.C

```diff
@@ -88,35 +116,74 @@ PatternedMeshGenerator::generate()
 
-  // Data structure that holds each row
-  _row_meshes.resize(_pattern.size());
+    const auto ids = MooseMeshUtils::getBoundaryIDs(*_meshes[i], boundary_names, false);
+    mooseAssert(ids.size() == boundary_names.size(),
+                "Unexpected number of ids returned for MooseMeshUtils::getBoundaryIDs");
 
-  // Getting the boundaries provided by the user
-  std::vector<BoundaryName> boundary_names = {getParam<BoundaryName>("left_boundary"),
-                                              getParam<BoundaryName>("right_boundary"),
-                                              getParam<BoundaryName>("top_boundary"),
-                                              getParam<BoundaryName>("bottom_boundary")};
+    // Keep track of used IDs so we can later find IDs that are unused across all meshes
+    used_boundary_ids.insert(ids.begin(), ids.end());
 
-  std::vector<boundary_id_type> ids =
-      MooseMeshUtils::getBoundaryIDs(*_meshes[0], boundary_names, true);
+    // Keep track of unique boundary ids seen in each input mesh
+    std::set<boundary_id_type> id_set;
+    // Keep track of indices of first instance of each unique boundary id
+    std::map<boundary_id_type, size_t> seen_bid_to_index_map;
+
+    size_t index = 0;
+    for (const auto side : make_range(4))
+    {
+      // Check if the boundary has been initialized
+      if (ids[side] == Moose::INVALID_BOUNDARY_ID)
+        paramError("inputs",
+                   "The '",
+                   boundary_param_names[side],
+                   "' parameter with value '",
+                   boundary_names[side],
+                   "' does not exist in input mesh '",
+                   _input_names[i],
+                   "'");
+
+      input_bids[i][side] = ids[side];
+
+      // We only do this when i == 0 because all input meshes should have the
+      // same index map. Allowing different index maps for different input
+      // meshes results in undefined behaviour when stitching
+      if (i == 0)
+      {
+        if (!id_set.count(ids[side]))
+        {
+          input_bids_unique[i].push_back(ids[side]);
+          id_set.insert(ids[side]);
+          seen_bid_to_index_map[ids[side]] = index;
+          boundary_name_2_index_map[boundary_param_names[side]] = index++;
+        }
+        else
+          boundary_name_2_index_map[boundary_param_names[side]] = seen_bid_to_index_map[ids[side]];
+      }
 
-  mooseAssert(ids.size() == boundary_names.size(),
-              "Unexpected number of ids returned for MooseMeshUtils::getBoundaryIDs");
+      else // i > 0
+      {
+        if (ids[side] != input_bids[i - 1][side])
+          have_common_ids = false;
+
+        if (!id_set.count(ids[side]))
+        {
+          input_bids_unique[i].push_back(ids[side]);
+          id_set.insert(ids[side]);
+        }
+      }
+    }
 
-  boundary_id_type left = ids[0], right = ids[1], top = ids[2], bottom = ids[3];
+    lengths_set.insert(input_bids_unique.size());
```

Comment:
I wouldnt keep track of length_sets like this.
If you make it a single integer, you can error as soon as you hit the inconsistent size, and you can return the name of the problematic inputs

Reply:
I will modify to use a single integer. However, I don't think there is a quick way to point to a single input mesh as the problematic input without adding additional checks. For example: if there are two input meshes with a different length of input_bids_unique, which one is considered the problematic input?

Reply:
the first one you hit in the loop? Or the first one that is not 1 or 4

Reply:
I guess it s why it s easier to check for 1/4 than to check for "all the same"